packages/astro/src/core/config/settings.ts

```diff
@@ -24,9 +24,16 @@ export function createBaseSettings(config: AstroConfig): AstroSettings {
 export function createSettings(config: AstroConfig, cwd?: string): AstroSettings {
 	const tsconfig = loadTSConfig(cwd);
 	const settings = createBaseSettings(config);
+
+	const watchFiles = tsconfig?.exists ? [tsconfig.path, ...tsconfig.extendedPaths] : [];
+
+	if(cwd) {
+		watchFiles.push(fileURLToPath(new URL('./package.json', pathToFileURL(cwd))));
+	}
```

Comment:
This seems like an odd edge case, but should we check if `package.json` exists first?

Since Deno is landing `npm:` support, I think we're not far off from an Astro project that doesn't have a `package.json` file being a feasible thing.

Reply:
We don't need to check for that. This is used that when a file does change and it matches one of the files in `watchFiles`, it will trigger a restart. If the file doesn't exist then it will never match. It doesn't cause some other process to try and read the (potentially not existing) files.

Here's the only place this information is used: https://github.com/withastro/astro/blob/7191ed1c7c80c9d70d7d6f88e40ebc9a89787411/packages/astro/src/core/dev/restart.ts#L55-L57